File: LowestCommonAncestor1/src/LowestCommonAncestor1.java
public class LowestCommonAncestor1 {

	/* Driver program to test LCA functions */
	public static void main(String args[])
	{
		DAG graph = new DAG();
		graph.root = new Node(1);
		graph.root.leftChild = new Node(2);
		graph.root.rightChild = new Node(3);
		graph.root.leftChild.leftChild = new Node(4);
		graph.root.leftChild.rightChild = new Node(5);
		graph.root.rightChild.leftChild = new Node(6);
		graph.root.rightChild.rightChild = new Node(7);

		Node lca = graph.findLCA(4, 5);
		if (lca != null)
			System.out.println("LCA(4, 5) = " + lca.data);
		else
			System.out.println("Keys are not present");

		lca = graph.findLCA(4, 10);
		if (lca != null)
			System.out.println("LCA(4, 10) = " + lca.data);
		else
			System.out.println("Keys are not present");
	}
}

File: LowestCommonAncestor1Python/LowestCommonAncestor1.py
from BinaryTree import BinaryTree
from Node import Node

class LowestCommonAncestor1:

        # Driver program to test above function
    def main():
        tree = BinaryTree()
        tree.root = Node(1)
        tree.root.leftChild = Node(2)
        tree.root.rightChild = Node(3)
        tree.root.leftChild.leftChild = Node(4)
        tree.root.leftChild.rightChild = Node(5)
        tree.root.rightChild.leftChild = Node(6)
        tree.root.rightChild.rightChild = Node(7)
        lca = tree.findLCA(tree.root, 4, 5)
        
        if lca is not None:
            print("LCA(4, 5) = ", lca.data)
        else:
            print("Node(s) are not present")

        lca = tree.findLCA(tree.root, 4, 10)
        if lca is not None:
            print("LCA(4,10) = ", lca.data)
        else:
            print("Node(s) are not present")

    if __name__ == '__main__': main()